generateBuildrootSBOM.py

```diff
@@ -35,38 +33,36 @@
 # Buildroot manifest.csv file header shows the following header row
 # PACKAGE,VERSION,LICENSE,LICENSE FILES,SOURCE ARCHIVE,SOURCE SITE,DEPENDENCIES WITH LICENSES
 #
+from cyclonedx.model.component import ComponentType
+
 
-def create_buildroot_sbom(args, parser: br_parser):
+def create_buildroot_sbom(args, br_bom):
+    br_bom_local: cyclonedx.model.bom = br_bom
     #
     # Capture the components that describe the complete inventory of first-party software
-    final_component_details = list("")
     # Buildroot CSV file supplies software package data in each row. Any change to that map of data will break
-    # the resulting JSON. Thus a try/except block to help with run time issues.
+    # the resulting JSON. Use a try/except block to help with run time issues.
     with open(args.input_file, newline='') as csvfile:
         sheetX = csv.DictReader(csvfile)
+        my_component_list = []
         for row in sheetX:
             try:
                 purl_info: str | Any = "pkg:generic/" + row['PACKAGE'] + "@" + row['VERSION'] + \
                                        "?download_url=" + row['SOURCE SITE'] + row['SOURCE ARCHIVE']
-                license_list_info = list("")
-                set_of_license_info = {"expression": row['LICENSE']}
-                license_list_info.append(set_of_license_info)
-                set_of_component_details = {"type": "library", "name": row['PACKAGE'], "version": row['VERSION'],
-                                            "licenses": license_list_info, "purl": purl_info}
-                final_component_details.append(set_of_component_details)
-
-                component_license = cyclonedx.model.License(name=row['LICENSE'])
-                component_license_choice: object = cyclonedx.model.LicenseChoice(license=component_license)
 
                 from packageurl import PackageURL
-                purl = PackageURL.from_string(purl_info)
+
                 componenttype = cyclonedx.model.component.ComponentType('firmware')
                 next_component = cyclonedx.model.component.Component(name=row['PACKAGE'],
                                                                      type=componenttype,
-                                                                     purl=purl,
-                                                                     licenses=[component_license_choice],
+                                                                     # component_type=componenttype,
+                                                                     # package_url_type=purl_info,
```

Comment:
you are lacking the purl?
previously purl was the only information there the package source was in.

please use Component's `external_references` to add a source-reference.
see https://cyclonedx-python-library.readthedocs.io/en/latest/autoapi/cyclonedx/model/index.html#cyclonedx.model.ExternalReference
i'd recommend using [ExternalReferenceType.DISTRIBUTION ](https://cyclonedx-python-library.readthedocs.io/en/latest/autoapi/cyclonedx/model/index.html#cyclonedx.model.ExternalReferenceType) as the externalReference's type.

Reply:
i'd recommend to have this change a separate pullrequest.